### PR TITLE
Add support for Oracle fault tolerant execution

### PR DESCRIPTION
**Pull request by chenjian2664**
## Additional context and related issues
Fixes https://github.com/trinodb/trino/issues/17199

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Oracle
* Add support for writes when [fault-tolerant
  execution](/admin/fault-tolerant-execution) is enabled. ({issue}`17200`)
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -571,6 +571,7 @@ jobs:
             - { modules: plugin/trino-mysql }
             - { modules: plugin/trino-mysql, profile: fte-tests }
             - { modules: plugin/trino-oracle }
+            - { modules: plugin/trino-oracle, profile: fte-tests }
```
[the remark this answers is not in the document]
Comment (chenjian2664) [reply .github/workflows/ci.yml:571]: Updated

[the remark this answers is not in the document]
Comment (losipiuk) [reply .github/workflows/ci.yml:571]: So didn't it make sense to combine fte and non-fte tests?

File: plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleTaskFailureRecoveryWithPoolEnabledTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.oracle;
+
+import io.trino.operator.RetryPolicy;
+
+public class TestOracleTaskFailureRecoveryWithPoolEnabledTest
```
Comment (losipiuk) [inline plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleTaskFailureRecoveryWithPoolEnabledTest.java:18]: How much do those take? I am not sure if we need full pool/no-pool task/query coverage

**Comment by chenjian2664** [comment]
https://github.com/trinodb/trino/actions/runs/4795033726/jobs/8534026056?pr=17200#step:6:8434
https://github.com/trinodb/trino/actions/runs/4795033726/jobs/8534026232?pr=17200#step:6:19665
Not relate to this change

File: plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleTaskFailureRecoveryWithPoolEnabledTest.java (hunk 2 above)
Comment (chenjian2664) [reply plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleTaskFailureRecoveryWithPoolEnabledTest.java:18]: Are suggest we could remove this(the pool enabled) test case?
> How much do those take?

I am not sure how to measure the cost, but I have done tests both situation two times on my local:

- Without pool enabled: times 13min42, 14min10, highest cpu usage consume (on docker container) around 2 cores, memory consumes around 500MB
- With pool enabled: times 13min20, 15min39, cpu and memory is similar to the pool disabled

File: plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleFailureRecoveryTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        return createOracleQueryRunner(
+                closeAfterClass(oracleServer),
+                configProperties,
+                coordinatorProperties,
+                ImmutableMap.<String, String>builder()
+                        .put("connection-url", oracleServer.getJdbcUrl())
+                        .put("connection-user", TEST_USER)
+                        .put("connection-password", TEST_PASS)
+                        .put("oracle.connection-pool.enabled", isConnectionPoolEnabled() ? "true" : "false")
+                        .put("oracle.remarks-reporting.enabled", "true")
```
Comment (hashhar) [inline plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleFailureRecoveryTest.java:60]: is this required for this test? remarks reporting allows table/column comments to be fetched but I don't think FTE tests use these. Enabling this makes some metadata operations slower.

File: plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleFailureRecoveryTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        TestingOracleServer oracleServer = new TestingOracleServer();
+        return createOracleQueryRunner(
+                closeAfterClass(oracleServer),
+                configProperties,
+                coordinatorProperties,
+                ImmutableMap.<String, String>builder()
+                        .put("connection-url", oracleServer.getJdbcUrl())
+                        .put("connection-user", TEST_USER)
+                        .put("connection-password", TEST_PASS)
+                        .put("oracle.connection-pool.enabled", isConnectionPoolEnabled() ? "true" : "false")
```
Comment (hashhar) [inline plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleFailureRecoveryTest.java:59]: FTE doesn't depend on connection pooling so no need to have two variants of this test. Let's just test without pooling (which is the default config).

File: plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleTaskFailureRecoveryWithPoolEnabledTest.java (hunk 2 above)
Comment (chenjian2664) [reply plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleTaskFailureRecoveryWithPoolEnabledTest.java]: @losipiuk Removed the pool enabled test cases. cc @hashhar

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hashhar) [reply .github/workflows/ci.yml:571]: Can be combined. It's 18m for normal test + 8min for FTE.

**Pull request merged**